File: pipeline/kite9-pipeline-common/src/main/java/com/kite9/k9server/adl/format/media/Format.java
package com.kite9.k9server.adl.format.media;

/**
 * Handles sending a certain file format to the output stream for http
 * responses.
 * 
 * @author robmoffat
 */
public interface Format {
	
	/**
	 * A canonical media type, used by Kite9 server consistently.
	 * Used for caching.
	 */
	public default String getFormatIdentifier() {
		return getMediaTypes()[0].toString();
	}

	public MediaType[] getMediaTypes();

	/**
	 * In order to use format=xxx in the URL, we need to give each format an extension.
	 */
	public String getExtension();
	
	
}

File: pipeline/kite9-pipeline-common/src/main/kotlin/com/kite9/k9server/adl/format/media/Format.kt
package com.kite9.k9server.adl.format.media

/**
 * Handles sending a certain file format to the output stream for http
 * responses.
 *
 * @author robmoffat
 */
interface Format {

    /**
     * A canonical media type, used by Kite9 server consistently.
     * Used for caching.
     */
    val formatIdentifier: String?
        get() = mediaTypes[0].toString()

    val mediaTypes: List<MediaType>

    /**
     * In order to use format=xxx in the URL, we need to give each format an extension.
     */
    val extension: String?
}